Include/Rover/Range.hpp

```diff
@@ -134,12 +134,14 @@ namespace Rover {
       return begin;
     }
     while(true) {
-      auto value = calculate_random(begin, end);
-      auto alt_value = calculate_random(begin, end);
+      auto value = calculate_random(begin, 2 * end);
```

Comment:
Not sure whether it will satisfy requirements for all custom types... We defined supported types as quotient spaces over doubles, but C++ built-in types do not technically work this way: if you multiply an integer and a double, you'll get a double rather than an integer. So I am relying on promotion of integer 2 to double here to support such corner cases. Let me know if you have better ideas.

Reply:
```end + end```